ground/src/main/proto/com/google/ground/job.proto

```diff
@@ -19,11 +19,189 @@ syntax = "proto3";
 option java_package = "com.google.android.ground.proto";
 option java_multiple_files = true;
 
-import "com/google/ground/style.proto";
-
+// A single job, defined by one or more tasks, and zero or more locations of
+// interest (aka data collection sites).
 message Job {
+    // UUID of the job.
     string id = 1;
-    string name = 2;
-    Style default_style = 3;
-    // TODO(#1748): Add `tasks` and `strategy` fields.
+    // Index of the job when shown in lists.
+    int32 index = 2;
+    // User-facing name of the job.
+    string name = 3;
+    // The style used when rendering geometries and other UI elements associated
+    // with this job.
+    Style default_style = 4;
+    // One or more data collection tasks to be carried out by the user in order
+    // to collect and submit data.
+    repeated Task tasks = 5;
+}
+
+// A style used when rendering geometries and other UI elements associated
+// with this job.
+message Style {
+    string color = 1;
 }
+
+// A data collection task to be carried out by the user/ to collect and submit
+// data.
+message Task {
+    // UUID of the task.
+    string id = 1;
+    // Index of the task when shown in lists.
+    int32 index = 2;
+    // Question or instructions shown to the user describing the task to be
+    // completed.
+    string prompt = 3;
+    // When true, the user may not skip this task.
+    bool required = 4;
+
+    // Defines possible values of the `level` field.
+    enum DataCollectionLevel {
+        // Default value when level is not specified.
+        UNSPECIFIED_DATA_COLLECTION_LEVEL = 0;
+        // Data collected at this level is used to define a location of
+        // interest (LOI). Specifically, geometry tasks of this type may be used
+        // to allow data collectors to define the LOI's geometry. Future
+        // implementations may use this to allow other LOI properties to be
+        // added or modified.
+        LOCATION_OF_INTEREST_METADATA = 1;
+        // Data collected at this level is associated with, but does not define,
+        // a location of interest (LOI). Specifically, geometry tasks of this
+        // type define one or more annotations of the associated LOI. rather
+        // than the geometry of the LOI itself.
+        LOCATION_OF_INTEREST_DATA = 2;
+    }
+    // Defines to which entity data collected in this task is associated with.
+    // Defaults to LOCATION_OF_INTEREST_DATA.
+    DataCollectionLevel level = 5;
+
+
+    // Conditions which control whether this task is shown during data
+    // collection. All conditions must be evaluate to `true` for the task to
+    // be triggered.
+    repeated Condition conditions = 6;
```

Comment:
Just for my understanding, it seems that we're simplifying the [currently implemented `Condition` logic](https://github.com/google/ground-android/blob/462d28b787ad9b49e3a970b0e0d933e9c0a89af2/ground/src/main/java/com/google/android/ground/model/task/Condition.kt#L31), which is fine. Currently we have a single Condition with a match type of three types: match one, any, or all, and then for each sub expression, an any, all or one option selected.

In this new model, we are hardcoding "all" match conditions then "any" sub choice. If I did my math right, I think we should be able to cover the "match_all" case by creating multiple conditions with one option in each condition, and the "match_any" case by creating one condition with multiple options in a single condition, and we'll forgo a "match_one" type.

Reply:
Yes, I didn't include those since they're not used and there aren't any FRs requesting them. I also think, as you said, that the various options can be represented sufficiently in this model. That said, additive changes to this model with reasonable defaults shouldn't break old clients, and I think the current model allows for adding those later without any structural changes (moving protos around in hierarchy etc).